packages/@aws-cdk/aws-applicationautoscaling/lib/scalable-target.ts

```diff
@@ -96,13 +96,14 @@ export class ScalableTarget extends Resource implements IScalableTarget {
   constructor(scope: Construct, id: string, props: ScalableTargetProps) {
     super(scope, id);
 
-    if (props.maxCapacity < 0) {
+    if (!Token.isUnresolved(props.maxCapacity) && props.maxCapacity < 0) {
```

Comment:
It took a while to understand this conditional because of the double negative. Could we instead add a `Token.isResolved()` method and use that instead, for better readability?

Gets a bit worse in `auto-scaling-group.ts`.